dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/SecretTool.java

```diff
@@ -70,7 +99,54 @@ public char[] getCharArraySystemSecret (final String key) {
 	public char[] getCharArraySystemSecret (final String key,
 								   final char[] defaultValue) {
 
+        if(!canUserEvaluate()){
+            throw new SecurityException("External scripting is disabled in your dotcms instance.");
+        }
 		final Optional<DotVelocitySecretAppConfig> config = DotVelocitySecretAppConfig.config(APILocator.systemHost());
 		return config.isPresent()? config.get().getCharArrayOrNull(key, defaultValue) : defaultValue;
 	}
+
+    private static final boolean ENABLE_SCRIPTING = Config.getBooleanProperty("ENABLE_SCRIPTING", false);
+
+    /**
+     * Test 2 things.
+	 * 1) see if the user has the scripting role
+	 * 2) otherwise check if the last modified user has the scripting role
+     * @return boolean
+     */
+    protected boolean canUserEvaluate() {
+
+        if(!ENABLE_SCRIPTING) {
+
+            Logger.warn(this.getClass(), "Scripting called and ENABLE_SCRIPTING set to false");
+            return false;
+        }
+
+        try {
+
+			boolean hasScriptingRole = false;
+			final User user = WebAPILocator.getUserWebAPI().getUser(this.request);
+			final Role scripting = APILocator.getRoleAPI().loadRoleByKey(Role.SCRIPTING_DEVELOPER);
+			// try with the current user
+			if (null != user) {
+
+				hasScriptingRole = APILocator.getRoleAPI().doesUserHaveRole(user, scripting);
```

Comment:
I think I would check this second, to prevent cache churn.  The majority of use cases will use the template to find out if it should be rendered